src/components/room-invitation-creation-modal.js

```diff
@@ -0,0 +1,78 @@
+import PropTypes from 'prop-types';
+import Logger from '../common/logger.js';
+import { Form, Modal, Input } from 'antd';
+import { useTranslation } from 'react-i18next';
+import errorHelper from '../ui/error-helper.js';
+import { useService } from './container-context.js';
+import React, { useState, useRef } from 'react';
+import RoomApiClient from '../api-clients/room-api-client.js';
+
+const FormItem = Form.Item;
+
+const logger = new Logger(import.meta.url);
+
+function RoomInvitationCreationModal({ isVisible, onClose, roomId }) {
+  const formRef = useRef(null);
+  const roomApiClient = useService(RoomApiClient);
+  const { t } = useTranslation('roomInvitationCreationModal');
+
+  const emailValidationRules = [
+    {
+      required: true,
+      message: t('emailRequired'),
+      whitespace: true
+    }
+  ];
+  const initialFormValues = { email: null };
+  const [loading, setIsLoading] = useState(false);
+
+  const onFinish = async values => {
+    try {
+      setIsLoading(prevState => ({ ...prevState, loading: true }));
+      await roomApiClient.addRoomInvitation({ email: values.email, roomId });
+      onClose(true);
+    } catch (error) {
+      errorHelper.handleApiError({ error, logger, t });
+    } finally {
+      setIsLoading(prevState => ({ ...prevState, loading: false }));
+    }
+  };
+
+  const handleCancel = () => {
+    if (formRef.current) {
+      formRef.current.resetFields();
+    }
+    onClose(false);
+  };
+
+  const submitForm = () => {
+    if (formRef.current) {
+      formRef.current.submit();
+    }
+  };
+
+  return (
+    <Modal
+      title={t('newRoomInvitation')}
+      onCancel={handleCancel}
+      onOk={submitForm}
+      maskClosable={false}
+      visible={isVisible}
+      okButtonProps={{ loading: loading.loading, htmlType: 'submit' }}
+      >
+      <Form name="new-room-invitation-form" initialValues={initialFormValues} onFinish={onFinish} ref={formRef} layout="vertical">
```

Comment:
I've gone for a programatic way to use the form here, if you like the approach this could be done for other forms that are used within modals. This has the advantage of getting rid of handling the state and form validation outside the form.